### ISSUE TITLE
Package Control: Error downloading channel. HTTP exception InvalidCertificateException

### ISSUE THREAD

**Issue by grych** [wbond/package_control#1246]
````
startup, version: 3126 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/grych/Library/Application Support/Sublime Text 3/Packages
state path: /Users/grych/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/grych/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.292219
using gpu buffer for window
first paint time: 0.343435
using gpu buffer for window
startup time: 0.37154
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin All Autocomplete.all_views_completions
reloading plugin Better CoffeeScript.CoffeeScript
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin Color Highlighter.ColorHighlighter
reloading plugin Color Highlighter.colors
reloading plugin DashDoc.DashDoc
reloading plugin Git.git_commands
reloading plugin GitGutter.git_gutter
reloading plugin GitGutter.git_gutter_compare
reloading plugin GitGutter.git_gutter_events
reloading plugin GitGutter.git_gutter_handler
reloading plugin GitGutter.git_gutter_jump_to_changes
reloading plugin GitGutter.git_gutter_popup
reloading plugin GitGutter.git_gutter_settings
reloading plugin GitGutter.git_gutter_show_diff
reloading plugin GitGutter.git_helper
reloading plugin GitGutter.promise
reloading plugin HTMLBeautify.HTMLBeautify
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SCSS.scss_completions
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SideBarEnhancements.Stats
reloading plugin BeautifyRuby.beautify_ruby
reloading plugin ColorPicker.sublimecp
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
plugins loaded
SublimeLinter: debug mode: off 
Package Control: Skipping automatic upgrade, last run at 2017-06-25 12:47:47, next run at 2017-06-25 13:47:47 or after
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
````

I've tried to add the certificate of packagecontrol.io, as well as Rapid SSL and GeoTrust to `Package Control.user-ca-bundle`, the same issue.

**Comment by keith-hall** [comment]
Have you tried the steps at https://packagecontrol.io/docs/troubleshooting#There_are_no_packages_available_for_installation? Maybe you are behind a proxy that rewrites secure connections? Enabling the debug logging may provide more info in the console: https://packagecontrol.io/docs/troubleshooting#Enabling_the_Debug_Log

**Comment by grych** [comment]
Yes, I've tried those, I added certificate to user-ca-bundle. I was not behind the proxy.

I will add the debug log when I am by the computer on which the error occured.

**Comment by wbond** [comment]
Can you try manually upgrading to Package Control 3.3.0 and see if that helps?

**Comment by grych** [comment]
All ok, thx!

**Comment by wbond** [comment]
It is working now?

**Comment by grych** [comment]
Yes, all seems to be working, thanks!

**Comment by Klemart3D** [comment]
Same problem for me with 3.3.0, without proxy :

```
Package Control: Skipping automatic upgrade, last run at 2017-10-15 14:53:38, next run at 2017-10-15 15:53:38 or after
Package Control: Fetching list of available packages and dependencies
  Platform: osx-x64
  Sublime Text Version: 3143
  Package Control Version: 3.3.0
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.34.243
  Resolved IPv6: 2600:3c02:e000:42::1
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/me/Library/Application Support/Sublime Text 3/Packages/User/oscrypto-ca-bundle.crt (268794 bytes)
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/me/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Using hostname "packagecontrol.io" for TLS SNI extension
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
```

**Comment by jonahpaul** [comment]
Same here.

``` Package Control: Fetching list of available packages and dependencies
  Platform: osx-x64
  Sublime Text Version: 3156
  Package Control Version: 3.3.0
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.34.243
  Resolved IPv6: 2600:3c02:e000:42::1
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/jonahpaul/Library/Application Support/Sublime Text 3/Packages/User/oscrypto-ca-bundle.crt (275262 bytes)
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/jonahpaul/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Using hostname "packagecontrol.io" for TLS SNI extension
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
Package Control: Fetching list of available packages and dependencies
  Platform: osx-x64
  Sublime Text Version: 3156
  Package Control Version: 3.3.0
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.34.243
  Resolved IPv6: 2600:3c02:e000:42::1
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/jonahpaul/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Using hostname "packagecontrol.io" for TLS SNI extension
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.
Package Control: No updated packages
environment variables loaded using: /bin/zsh -l```

**Comment by akhats** [comment]
I updated Sublime text today with the latest build but that cause Package Control to fail again on Mac OS Sierra (10.13.3). I remember it used to fail earlier also but there was a workaround. Below is the console output I got
`tartup, version: 3170 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/arun/Library/Application Support/Sublime Text 3/Packages
state path: /Users/arun/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/arun/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.272572
using gpu buffer for window
startup time: 0.349406
environment variables loaded using: /bin/bash -l
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin 0_package_control_loader.00-package_control
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
plugins loaded
Package Control: Skipping automatic upgrade, last run at 2018-05-10 01:31:53, next run at 2018-05-10 02:31:53 or after
Package Control: Fetching list of available packages and dependencies
  Platform: osx-x64
  Sublime Text Version: 3170
  Package Control Version: 3.3.0
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.34.243
  Resolved IPv6: ::ffff:50.116.34.243
Package Control: Urllib Debug Proxy
  http_proxy: 
  https_proxy: 
  proxy_username: 
  proxy_password: 
Package Control: Found previously exported CA bundle at /Users/arun/Library/Application Support/Sublime Text 3/Packages/User/oscrypto-ca-bundle.crt (275197 bytes)
Package Control: Urllib HTTPS Debug General
  Connecting to packagecontrol.io on port 443
Package Control: Urllib HTTPS Debug General
  Upgrading connection to SSL using CA certs file at /Users/arun/Library/Application Support/Sublime Text 3/Packages/User/Package Control.merged-ca-bundle
  Using hostname "packagecontrol.io" for TLS SNI extension
Package Control: Error downloading channel. HTTP exception InvalidCertificateException (Host packagecontrol.io returned an invalid certificate ([SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:548))) downloading https://packagecontrol.io/channel_v3.json.`
